File: euler_047.py
from time import time
from primePy import primes
from collections import Counter
from itertools import count


def pfs(n):
    return set(f for f in Counter(primes.factors(n)).most_common())

def four_cons(n):
    res = pfs(n)
    for i in range(1,4):
        res = res.union(pfs(n + i))
    return res

start = time()

for n in count(1):
    if len(four_cons(n)) == 16:
        print("First of four consecutive numbers: " + str(n))
        break


stop = time()
print("Time: " + str(stop-start))

File: euler_047.js
const start = Date.now();

// Primes
const upperBound = 200000;

let sieve = Array(upperBound);
sieve.fill(true);
sieve[0] = false;
sieve[1] = false;

let primes = [];

for (let c = 2;c < upperBound; c++) {
    if (sieve[c]) {
        primes.push(c);
        let j = 2;
        while (c * j < upperBound) {
            sieve[c * j] = false;
            j++;
        }
    }
}
// End of primes

let numberOfPrimeFactors = Array(upperBound);
numberOfPrimeFactors.fill(0);

for (const p of primes) {
    numberOfPrimeFactors[p] = 1;
    let k = 2;
    while (p * k < upperBound) {
        numberOfPrimeFactors[p * k]++;
        k++;
    }
}

let res = 1;
let stillLooking = true;

while (stillLooking) {
    if (numberOfPrimeFactors[res] === 4) {
        let count = 1;
        for (let i = 1; i < 4; i++) {
            if (numberOfPrimeFactors[res + i] === 4) {
                count++;
            }
        }
        if (count === 4) {
            stillLooking = false;
        }
    }
    res++;
}





const end = Date.now();

console.log("Execute time: ", end - start, " The first of the numbers is: ", res - 1);